src/lib/__tests__/chartStorage.js

```diff
@@ -0,0 +1,310 @@
+const { JSDOM } = require('jsdom'); 
+
+// Mock the Local Storage. 
+
+const dom = new JSDOM('<!doctype html><html><body></body></html>', {
+    url: 'http://localhost'
+});
+global.window = dom.window;
+global.document = dom.window.document;
+global.localStorage = {
+    getItem: jest.fn(),
+    setItem: jest.fn(),
+}; 
+const { saveChart , loadAllSavedCharts , loadSavedChart , updateCurrentChartData , loadCurrentChartData } = require( "../chartStorage" ) ; 
+
+// Function 1. 
+
+// If an index is not provided in the input. 
+
+test('Expect, if there is no index input, saveChart to save the chart. ', () => { 
+    window.localStorage.clear() ; 
+    // Create the data. 
+    const chartData = {
+        type: 'line',
+        data: [
+            { x: 1, y: 10 },
+            { x: 2, y: 20 },
+            { x: 3, y: 30 }
+        ],
+        xLabel: 'X Axis',
+        yLabel: 'Y Axis',
+        title: 'Sample Chart',
+        color: '#FFFFFF'
+    }; 
+    // Call the function to test. 
+    saveChart(chartData); 
+    // Check if the output is correct. 
+    const savedCharts = JSON.parse(window.localStorage.getItem('savedCharts'));
+    expect(savedCharts).toHaveLength(1); 
+    expect(savedCharts[0]).toEqual(chartData); 
+    window.localStorage.clear() ; 
+}); 
+
+// If an index is provided in the input. 
+
+test('Expect, if there is an index input, saveChart to save the chart to the correct index. ', () => { 
+    window.localStorage.clear() ; 
+    // Create the data. 
+    const chart_1 = {
+        type: 'line',
+        data: [
+            { x: 1, y: 10 },
+            { x: 2, y: 20 },
+            { x: 3, y: 30 }
+        ],
+        xLabel: 'X Axis',
+        yLabel: 'Y Axis',
+        title: 'Sample Chart',
+        color: '#FFFFFF'
+    }; 
+    const chart_2 = {
+        type: 'line',
+        data: [
+            { x: 3, y: 30 },
+            { x: 2, y: 20 },
+            { x: 1, y: 10 }
+        ],
+        xLabel: 'X Axis',
+        yLabel: 'Y Axis',
+        title: 'Sample Chart',
+        color: '#FFFFFF'
+    }; 
+    // Create the setup. 
+    saveChart(chart_1); 
+    // Call the function to test. 
+    saveChart(chart_2 , 0 ); 
+    // Check if the output is correct. 
+    const savedCharts = JSON.parse(window.localStorage.getItem('savedCharts'));
+    expect(savedCharts).toHaveLength(1); 
+    expect(savedCharts[0]).toEqual(chart_2); 
+    window.localStorage.clear() ; 
+}); 
+
```

Comment:
Maybe add a test for when the index is greater than charts.length? Just to determine that the chart is correctly pushed to the storage, at the end of the already existing charts.